### PR TITLE
Adding formatting on save for ts files

### LINKED ISSUE #24641: [TypeSpecValidation] Enable formatOnSave in VSCode

### PR DESCRIPTION
**Pull request by ckairen**
closes #24641

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -23,5 +23,6 @@
   "typescript.tsdk": "node_modules\\typescript\\lib",
   "[json]": {
     "editor.defaultFormatter": "esbenp.prettier-vscode"
-  }
+  },
```
Comment (mikeharder) [inline .vscode/settings.json:26]: Can we enable this at a smaller scope, say only for the `eng` folder but not the rest of the repo?

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: .prettierrc.json [hunk 2]
```diff
@@ -27,6 +27,16 @@
...
       ]
+    },
+    {
+      "files": [
+        "**/*.ts"
+      ],
+      "options": {
+        "printWidth": 80,
+        "endOfLine": "auto",
+        "tabWidth": 4
```
Comment (mikeharder) [inline .prettierrc.json:38]: I recommend copying the settings from the typespec repo:

https://github.com/microsoft/typespec/blob/799cb6af21b5274543701096af8ecb9d5c9bb793/.prettierrc.json#L2-L8

File: eng/tools/TypeSpecValidation/src/TypeSpecValidation.ts [hunk 3]
```diff
@@ -1,64 +1,80 @@
-import {exec} from "child_process";
```
Comment (mikeharder) [inline eng/tools/TypeSpecValidation/src/TypeSpecValidation.ts:1]: You will need to merge this file from main and re-format.

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+**/*.ts
```
Comment (mikeharder) [inline .prettierignore:1]: After further consideration, I would delete this file completely, and apply formatting to all `*.ts` files in the repo.  There's only one other such file (`scripts/prettier-swagger-plugin.ts`), and I don't mind if it gets formatted.

Also, VSCode shows this UI when saving an excluded file (the highlighted `X Prettier`), which I don't think users will like:

image

**Review by mikeharder — APPROVED** [review]
Approved pending changes in my comments

**Review by benbp — APPROVED** [review]

**Pull request merged**